main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -679,6 +703,30 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       SP ~ "(" ~ optWS ~ zeroOrMore(Expression).separatedBy(optWS ~ "," ~ optWS) ~ optWS ~ ")" ~ SP ~> ParsedAst.Expression.Tuple
     }
 
+    def ArrayLit: Rule1[ParsedAst.Expression] = rule {
+      SP ~ atomic("[|") ~ optWS ~ zeroOrMore(Expression).separatedBy(optWS ~ "," ~ optWS) ~ optWS ~ atomic("|]") ~ SP ~> ParsedAst.Expression.ArrayLit
```

Comment:
ReasonML uses `[| |]` for arrays so that makes sense. But I wonder if we should use `[1, 2, 3]` for arrays and perhaps `[|1, 2, 3|]` for vectors?

Reply:
Would be that even be possible?